bson/bsoncodec/string_codec_test.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (C) MongoDB, Inc. 2017-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package bsoncodec
+
+import (
+	"reflect"
+	"testing"
+
+	"go.mongodb.org/mongo-driver/bson/bsonoptions"
+	"go.mongodb.org/mongo-driver/bson/bsonrw/bsonrwtest"
+	"go.mongodb.org/mongo-driver/bson/bsontype"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/internal/testutil/assert"
+)
+
+func TestStringCodec(t *testing.T) {
+	t.Run("ObjectIDAsHex", func(t *testing.T) {
+		oid := primitive.NewObjectID()
+		reader := &bsonrwtest.ValueReaderWriter{BSONType: bsontype.ObjectID, Return: oid}
+		testCases := []struct {
```

Comment:
it might be easier if you had the input and the expected output inside the test case. Then running the tests doesn't have any if conditions in them... just, did you get the output you expected?

Reply:
done